tailcfg/tailcfg.go

```diff
@@ -875,10 +875,21 @@ type Hostinfo struct {
 	// explicitly declared by a node.
 	Location *Location `json:",omitempty"`
 
+	TPM *TPMInfo `json:",omitempty"` // TPM device metadata, if available
+
 	// NOTE: any new fields containing pointers in this type
 	//       require changes to Hostinfo.Equal.
 }
 
+// TPMInfo contains information about a TPM 2.0 device present on a node.
+type TPMInfo struct {
+	Manufacturer    string `json:",omitempty"`
+	Vendor          string `json:",omitempty"`
+	Model           int    `json:",omitempty"`
+	FirmwareVersion uint64 `json:",omitempty"`
+	SpecRevision    int    `json:",omitempty"`
```

Comment:
give example values for all these in comments

FirmwareVersion is a uint64?

Reply:
Added doc comments and spec references.

The `FirmwareVersion` is weird, it ends up being `1688862745231975` on my laptop. Here's how [the spec](https://trustedcomputinggroup.org/wp-content/uploads/Trusted-Platform-Module-2.0-Library-Part-2-Version-184_pub.pdf) defines it:
* TPM_PT_FIRMWARE_VERSION_1 the most-significant 32 bits of a TPM vendor-specific value indicating the version number of the firmware. See Clause 10.12.2 and Clause 10.12.12.
* TPM_PT_FIRMWARE_VERSION_2 the least-significant 32 bits of a TPM vendor-specific value indicating the version number of the firmware. See Clause 10.12.2 and Clause 10.12.12.

I honestly don't know if `FirmwareVersion` is useful, happy to drop it.